### ISSUE TITLE
Why am I getting expected string, got function?

### ISSUE THREAD

**Issue by faceyspacey** [zimme/meteor-active-route#35]
I'm getting this issue all of a sudden when visiting a new route I just added:

````
Exception in template helper: Error: Match error: Expected string, got function in field ["class"]
````

How could this be?

**Comment by faceyspacey** [comment]
it seems to be caused by using findOne() instead of find() as the returns of iron router ```data``` handlers. The issue went away when i changed it to a plain find() call.

**Comment by zimme** [comment]
The object returned from the database when using findOne, most likely had a member named 'class" which was a function. The helpers now uses the data context to populate the options which is most likely the cause of this error message.

**Comment by akiarostami** [comment]
I'm getting error too.  I'm using 

`{{#if isNotActivePath regex='login|password'}}`

and I'm getting this error:

`Exception in template helper: Error: Match error: Expected string, got undefined in field ["class"]`

**Comment by faceyspacey** [comment]
@zimme that is exactly the issue! ...is there another options? the 'class' property cant change for me--what are my other options to avoid this?

**Comment by zimme** [comment]
On vacation right now in magaluf but when i get back ill think about skipping the data context inheritance in favour of a data attribute the way dynamic template does it

**Comment by zimme** [comment]
I'm not sure what's the best approach to this "problem" is.
1. Skipping the data context inheritance and inherit from the data attribute instead.
2. Adding an option `className` which when set to `true` will use `className` instead of `class`, this would remove the `class`/`className` alias which is in place right now.

**Comment by faceyspacey** [comment]
i have both ```class``` and ```className``` properties on my objects. So for me only the fist option would work. It does seem precarious relying on the data context for the route.

**Comment by zimme** [comment]
The reason I added this was to mimic the behaviour of `{{pathFor ...}}` and `{{linkTo ...}}` helpers which `iron:router` provides.

I think I'll just add support for the `data` attribute and an option to disable data context inheritance.

Sound good?

**Comment by zimme** [comment]
or maybe it's better to have data context inheritance as opt-in instead? but Meteor do have implicit data context inheritance by default for templates.

**Comment by faceyspacey** [comment]
meteor does, but urs is more of an automated solution, which now is assuming maybe too much about people's code. it's probably best to keep it all to what's specifically provided as parameters.

**Comment by zimme** [comment]
I haven't forgotten about this, I've just been busy with work.

My thinking is that I'll skip the data context inheritance and inherit from `data` attribute just like `Template.dynamic` does.

As this is a breaking change I'll release a new major version when this is done, I'll be leaving support for `meteorhacks:flow-router` behind with this release too.

**Comment by jorisroling** [comment]
I run into this because I have a **path** member. Please make these kind of assumptions about the context an **opt-in**, or have a decent **opt-out**.

**Comment by zimme** [comment]
Hopefully I can get this done tonight, if not it should be done by the end of the week.

**Comment by jorisroling** [comment]
Would be grand!

**Comment by zimme** [comment]
This should be fixed with version `2.3.1`. Please ping me here if it's not resolved and I'll reopen this issue.

**Comment by jorisroling** [comment]
Works... Thanks!
